### PR TITLE
Add token introspection requirements, example

### PR DESCRIPTION
**Pull request by jmandel**
Since ONC has required Token Introspection in its 2020 final rule, it's helpful to provide guidance and requirements that steer the community toward consistent implementation.

### CODE REVIEW

**Comment by isaacvetter** [comment]
Relevant section from spec:

>   To prevent token scanning attacks, the endpoint MUST also require
   some form of authorization to access this endpoint, such as client
   authentication as described in OAuth 2.0 [RFC6749] or a separate
   OAuth 2.0 access token such as the bearer token described in OAuth
   2.0 Bearer Token Usage [RFC6750].  The methods of managing and
   validating these authentication credentials are out of scope of this
   specification.

https://tools.ietf.org/html/rfc7662#section-2.1

File: spec/token-introspection/index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Required fields in the introspection response
+
+In addition to the `active` field required by RFC7662 (a boolean indicating whether the access token is active), the following fields SHALL be included in the introspection response:
+
+* `scope`. As included in the original access token response. The list of scopes granted by the authorization server as a space-separated JSON string.
+
+* `client_id`. As included in the original access token response. The client identifier of the client to which the token was issued.
+
+* `exp`. As included in the original access token response. The integrer timestamp indicating when the access token expires.
```
Comment (isaacvetter) [inline spec/token-introspection/index.md:16]:
```suggestion
* `exp`. As included in the original access token response. The integer timestamp indicating when the access token expires.
```

File: fsh/ig-data/input/pages/token-introspection.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+     {
+      "active": true,
+      "client_id": "07a89bd2-52ce-4576-8b85-71698efa8328",
+      "scope": "patient/*.read openid fhirUser",
+      "sub": "c91dfe96-731d-4e66-b4f9-01f8f8a4b7b2",
+      "patient": "Patient/456",
+      "fhirUser": "Patient/123",
```
Comment (rsmayda) [inline fsh/ig-data/input/pages/token-introspection.md:62]: Out of curiosity what is the use-case where the the requestor's `patient` claim is different than its `fhirUser` claim?

Comment (isaacvetter) [reply fsh/ig-data/input/pages/token-introspection.md:62]: Hey @rsmayda, it's proxies, such as parents or other caregivers authorizing an app to access data about another patient. It's an edge-case, but not uncommon.

Comment (rsmayda) [reply fsh/ig-data/input/pages/token-introspection.md:62]: Perfect, that answers that. Thanks!

**Pull request merged**